featurebyte/query_graph/sql/adapter/snowflake.py

```diff
@@ -280,3 +286,108 @@ def get_percentile_expr(cls, input_expr: Expression, quantile: float) -> Express
+                )
+            )
+
+        # Update agg_exprs select keys to use the aliases from the inner join subquery
+        groupby_subquery_alias = "GROUP_BY_RESULT"
+        new_groupby_exprs = []
+        for agg_expr in agg_exprs:
+            new_groupby_exprs.append(
+                alias_(
+                    get_qualified_column_identifier(agg_expr.alias, groupby_subquery_alias),
+                    alias=agg_expr.alias,
+                    quoted=True,
+                )
+            )
+
+        # Initialize the first join that has the initial select.
+        table_alias = cls._get_groupby_table_alias(0)
+        vector_expr = vector_aggregate_columns[0].aggr_expr.subquery(alias=table_alias)
+        # Rename the TABLE that we're selecting the keys from to be the aggregated table, aliased by the subquery above.
+        renamed_table_select_keys = []
+        for select_key in select_keys:
+            renamed_table_select_keys.append(
+                alias_(
+                    get_qualified_column_identifier(select_key.alias, table_alias),
+                    alias=select_key.alias,
+                    quoted=True,
+                )
+            )
+        left_expression = select(
+            *renamed_table_select_keys, *new_groupby_exprs, *vector_agg_select_keys
+        ).from_(vector_expr)
+        # Chain the remaining joins with the remaining vector aggregates if there are more than one
+        for idx, vector_agg_expr in enumerate(vector_aggregate_columns[1:]):
+            right_expr = vector_agg_expr.aggr_expr.subquery(
+                alias=cls._get_groupby_table_alias(idx + 1)
+            )
+            join_conditions = []
+            for select_key in select_keys:
+                join_conditions.append(
+                    expressions.EQ(
+                        this=get_qualified_column_identifier(
+                            select_key.alias, cls._get_groupby_table_alias(idx)
+                        ),
+                        expression=get_qualified_column_identifier(
+                            select_key.alias, cls._get_groupby_table_alias(idx + 1)
+                        ),
+                    )
+                )
+            left_expression = left_expression.join(
+                right_expr,
+                join_type="INNER",
+                on=expressions.and_(*join_conditions),
+            )
+
+        # Join with normal aggregation groupby's if there are any.
+        if agg_exprs:
+            join_conditions = []
+            for select_key in select_keys:
+                join_conditions.append(
+                    expressions.EQ(
+                        this=get_qualified_column_identifier(
+                            select_key.alias, groupby_subquery_alias
+                        ),
+                        expression=get_qualified_column_identifier(
+                            select_key.alias, f"T{len(vector_aggregate_columns) - 1}"
```

Comment:
Should the table alias be generated by `_get_groupby_table_alias`?

Reply:
ooh good spot. 

will add an integration test in a follow-up for that case too.